dbms/src/Common/Exception.cpp

```diff
@@ -24,7 +24,17 @@ namespace ErrorCodes
 void throwFromErrno(const std::string & s, int code, int e)
 {
 	char buf[128];
+#ifndef _GNU_SOURCE
+	if (!strerror_r(e, buf, sizeof(buf))) {
+		std::string tmp = std::to_string(code);
+		const char* code = tmp.c_str();
+		strcpy(buf, "Unknown error ");
+		strcpy(buf+13, code);
```

Comment:
Thx, will fix that.

strerror is available on MacOS, but it doesn't return the error string, instead it fills the buffer with the message.